### ISSUE TITLE
swipe issue when used SwipeRefreshLayout

### ISSUE THREAD

**Issue by luminousman** [h6ah4i/android-advancedrecyclerview#172]
<android.support.v4.widget.SwipeRefreshLayout  xmlns:android="http://schemas.android.com/apk/res/android"
            android:id="@+id/mSwipeRefreshLayout"
            android:layout_width="match_parent"
            android:layout_height="match_parent">

                <android.support.v7.widget.RecyclerView
                    android:id="@+id/recyclerSearchResult"
                    style="@style/myRecyclerViewStyle"
                    android:layout_width="match_parent"
                    android:layout_height="match_parent"
                    android:padding="5dp"
                    tools:listitem="@layout/my_conn_item" />

        </android.support.v4.widget.SwipeRefreshLayout>

When you swipe left or right. It does not swipe some time and released to native position while swiping. It's happen when you have used any vertical scroll also like `SwipeRefreshLayout`

**Comment by h6ah4i** [comment]
Thanks for your report. I will take it a look.

**Comment by livotov** [comment]
Yep, can confirm the same. Basically, when you're trying to swipe an item with the SwipeRefreshLayout installed into the same activity/fragment, item swipe action becomes cancelled on any non strictly horizontal finger movement (even small one) - looks like SwipeRefreshLayout intercepts the events and starts processing pull to refresh gesture

**Comment by sameerror** [comment]
Any workarounds to this issue? I tried disabling and enabling the SwipeRefreshLayout in the adapter onSwipeItem callback, works to some extent

**Comment by sameerror** [comment]
this worked for me:

implemented RecyclerViewSwipeManager.OnItemSwipeEventListener on the fragment containing the SwipeRefreshLayout and in the callbacks disable and enabled swipeRefreshLayout
`
    public void onItemSwipeStarted(int position) {
       mSwipeRefreshLayout.enableSwipe(false);
    }

    public void onItemSwipeFinished(int position, int result, int afterSwipeReaction) {
       mSwipeRefreshLayout.enableSwipe(true);
    }
`

**Comment by ma123** [comment]
sameerror, do you have custom SwipeRefreshLayout? If so, please as it looks. Thanks.

**Comment by ma123** [comment]
I used this solution and it workd. [http://stackoverflow.com/questions/31226658/swiperefreshlayout-interferes-with-scrollview-in-a-viewpager](url)

**Comment by joshfriend** [comment]
Perhaps `RecyclerViewTouchActionGuardManager` should have a `attachSwipeRefresh` method similar to `attachRecyclerView` which disables the swipe refresh view when an item interaction is taking place.
